Therify/directory
# feat: in app notifications

# Description
Adds firebase integration from previous NX app.
Adds: 
- Firebase and Firebase Admin vendors
- Notification service with create and read methods
- In-app Notification type to `types` lib
- Notifications Context and Provider.
- `useInAppNotificationDrawer` hook to handle drawer notification state and interactions
- Feature schema for notification service

# Closes issue(s)
NA
# How to test / repro
- Spin up app locally.
- Login and get therify user Id
- Create a test route that calls the `notificationsService.inApp.create` method passing the user id as the `targetUserId` prop
- See the notification count increase in the frontend app

# Screenshots
#### Notification Drawer
![image](https://user-images.githubusercontent.com/25045075/215562941-646f01b6-7576-4ec4-9957-9b698b250587.png)
#### Shows unread navigation action in sidebar menu
![image](https://user-images.githubusercontent.com/25045075/215548413-3edaa14a-edfb-4b2c-81dc-5aebff232c6a.png)
#### Shows unread navigation action in mobile menu
![image](https://user-images.githubusercontent.com/25045075/215548756-7a571dec-429e-45ee-b3a9-7b1b74588354.png)


# Changes include


# Checklist


# Other comments
